[i18n] change next/router for infra/i18n (part. 5)

richardaum opened the pull request:
### Goals

- Replace next/router for infra/i18n for some redirects on /play, /room, /g.
- Replace next/router for infra/i18n for SEO Breadcrumbs (because of pathname).
- Replace onRouteChangeComplete for a subscription of a listener on a specific event: `routeChangeComplete` (see https://stackoverflow.com/a/53933774).
- Translate shareable room link by making `translateHref` function also call `lngPathCorrector` - which is an internal function from next-i18next.

These changes should finish all required changes to make navigation compliant to i18n :tada:

vdfdev commented:
Interesting enough, `translateHref` does not seem to be adding the /en/ on the room page:

![image](https://user-images.githubusercontent.com/1034631/116775053-74782c80-aa15-11eb-90c4-5d03b6a4d31b.png)

vdfdev commented:
So the clipboard still misses /en/
![image](https://user-images.githubusercontent.com/1034631/116775083-b5704100-aa15-11eb-806b-26f7597dcea8.png)

I can try to help you debugging it tomorrow... I am going to sleep now :P

richardaum commented:
I think now it is fine... Last change I made I introduced a minor bug. I think it is now fixed.

![Captura de tela de 2021-05-01 12-01-45](https://user-images.githubusercontent.com/2018161/116786469-26931100-aa75-11eb-90fe-e902b9642dc2.png)

vdfdev commented:
It worked and looks great! Merging now, and then I will try to flip the flag in staging and then prod.
